File: platforms/kotlin/examples/scene/light/set_position.kt
import org.fragmentcolor.*

val lamp = Light.point(listOf(0.0f, 0.0f, 0.0f), listOf(1.0f, 1.0f, 1.0f))
lamp.setPosition(listOf(3.0f, 1.5f, -2.0f))

// Directional lights have no position — the call errors.
val sun = Light.directional(listOf(0.0f, -1.0f, 0.0f), listOf(1.0f, 1.0f, 1.0f))
val result = sun.setPosition(listOf(0.0f, 0.0f, 0.0f))

File: platforms/python/examples/scene/light/set_position.py
from fragmentcolor import Light

lamp = Light.point([0.0, 0.0, 0.0], [1.0, 1.0, 1.0])
lamp.set_position([3.0, 1.5, -2.0])

# Directional lights have no position — the call errors.
sun = Light.directional([0.0, -1.0, 0.0], [1.0, 1.0, 1.0])
result = sun.set_position([0.0, 0.0, 0.0])